### PR TITLE
improve asString performance

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/serializer.js [hunk 1]
```diff
@@ -87,23 +87,22 @@ module.exports = class Serializer {
...
-      str = str.source
-    } else if (typeof str !== 'string') {
-      str = str.toString()
+    if (typeof str !== 'string') {
+      if (str === null) {
+        return '""'
+      } else if (str instanceof Date) {
+        return '"' + str.toISOString() + '"'
+      } else if (str instanceof RegExp) {
+        str = str.source
```
Comment (RafaelGSS) [inline lib/serializer.js:96]:
```suggestion
      if (str === null) {
        return '""'
      }
      if (str instanceof Date) {
        return '"' + str.toISOString() + '"'
      } 
      if (str instanceof RegExp) {
        str = str.source
```

Just a nitpick to improve the legibility.

**Review by RafaelGSS — APPROVED** [review]
LGTM

**Review by ivan-tymoshenko — APPROVED** [review]
lgtm

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**